aleph_alpha_client/completion.py

```diff
@@ -0,0 +1,116 @@
+
+        logit_bias (dict mapping token ids to score, optional, default None)
+            The logit bias allows to influence the likelihood of generating tokens. A dictionary mapping token ids (int) to a bias (float) can be provided. Such bias is added to the logits as generated by the model.
+
+        log_probs (int, optional, default None)
+            Number of top log probabilities to be returned for each generated token. Log probabilities may be used in downstream tasks or to assess the model's certainty when producing tokens.
+
+        stop_sequences (List(str), optional, default None)
+            List of strings which will stop generation if they're generated. Stop sequences may be helpful in structured texts.
+
+            Example: In a question answering scenario a text may consist of lines starting with either "Question: " or "Answer: " (alternating). After producing an answer, the model will be likely to generate "Question: ". "Question: " may therfore be used as stop sequence in order not to have the model generate more questions but rather restrict text generation to the answers.
+
+        tokens (bool, optional, default False)
+            return tokens of completion
+
+        disable_optimizations  (bool, optional, default False)
+            We continually research optimal ways to work with our models. By default, we apply these optimizations to both your prompt and  completion for you.
+
+            Our goal is to improve your results while using our API. But you can always pass disable_optimizations: true and we will leave your prompt and completion untouched.
+    """
+
+    prompt: Union[str, List[Union[str, ImagePrompt]]] = ""
+    maximum_tokens: Optional[int] = 64
+    temperature: Optional[float] = 0.0
+    top_k: Optional[int] = 0
+    top_p: Optional[float] = 0.0
+    presence_penalty: Optional[float] = 0.0
+    frequency_penalty: Optional[float] = 0.0
+    repetition_penalties_include_prompt: Optional[bool] = False
+    use_multiplicative_presence_penalty: Optional[bool] = False
+    best_of: Optional[int] = None
+    n: Optional[int] = 1
+    logit_bias: Optional[Dict[int, float]] = None
+    log_probs: Optional[int] = None
+    stop_sequences: Optional[List[str]] = None
+    tokens: Optional[bool] = False
+    disable_optimizations: Optional[bool] = False
+
+    def render_as_body(self, model: str, hosting: str) -> Dict[str, Any]:
+        return {
+            "model": model,
+            "hosting": hosting,
+            "prompt": _to_serializable_prompt(self.prompt),
+            "maximum_tokens": self.maximum_tokens,
+            "temperature": self.temperature,
+            "top_k": self.top_k,
+            "top_p": self.top_p,
+            "presence_penalty": self.presence_penalty,
+            "frequency_penalty": self.frequency_penalty,
+            "best_of": self.best_of,
+            "n": self.n,
+            "logit_bias": self.logit_bias,
```

Comment:
? A dict int -> float does not render as valid json, does it?

Reply:
I think it used to be that way and that's why I did not question that. But I added a test for that anyway and it seems to work.